src/launch/github/generate_github_token.py

```diff
@@ -0,0 +1,100 @@
+import sys
+import time
+from pathlib import Path
+
+import boto3
+import requests
+from botocore.exceptions import ClientError
+from jwt import PyJWT
+
+
+def create_jwt(
+    application_id: int, private_key: bytes, expiration_seconds: int = 600
```

Comment:
In the original code I sent you, I chose 600 seconds as the default here because it made sense for my use case.

I'm pretty sure it will be fine for 99% of pipeline runs, but I can also see a scenario where we need to use the token and have a particularly long-running stage, maybe running tests of some sort. If the token expires during the test run and we try to use it later in the stage it would cause a failure.

I would consider adding a `--token-expiration-seconds` option to the entrypoint and the necessary wiring to pass this value to this method. Per the docs here, the [max lifetime is 8 hours](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/token-expiration-and-revocation#user-token-expired-due-to-github-app-configuration), so we shouldn't accept values higher than that.